permissionsdispatcher-processor/src/main/java/permissions/dispatcher/processor/JavaFileBuilder.java

```diff
@@ -74,12 +74,26 @@ private static MethodSpec createOnRequestPermissionsResult(ClassName target, Lis
                 .addParameter(int[].class, "grantResults")
                 .returns(void.class)
                 .beginControlFlow("switch (requestCode)");
-        for (ExecutableElement method : methods) {
+        for (ExecutableElement method : permissionMethods) {
             String methodName = method.getSimpleName().toString();
             methodBuilder
                     .addCode("case $N:\n", getRequestCodeFieldName(methodName))
                     .beginControlFlow("if ($T.verifyPermissions(grantResults))", PERMISSION_UTILS)
-                    .addStatement("target.$N()", methodName)
+                    .addStatement("target.$N()", methodName);
+            NeedsPermission needsPermission = method.getAnnotation(NeedsPermission.class);
+            String needsPermissionValue = needsPermission != null ? needsPermission.value() : "";
```

Comment:
I ran into an NPE while working on the PR earlier, though. I think that might be related to the library providing two Annotations expressing the need for permissions, @NeedsPermission and @NeedsPermissions. I will look into this again